sculptor-version/src/main/kotlin/io/papermc/sculptor/version/SculptorVersion.kt

```diff
@@ -169,7 +176,10 @@ class SculptorVersion : Plugin<Project> {
 
             mainClass = "net.minecraft.server.Main"
 
-            args("--nogui")
+            // default args, overridable in build.gradle
+            (if (mache.runServerArgs.getOrElse(listOf()).isEmpty()) listOf(
```

Comment:
Should we not check if the args contain the flag rather than if it's empty? Or is it intentional so if you override it you can still get the gui?

Reply:
I think all the getOrElse in this PR should be replace with defaults on the mache extension (so that the defaults are all in one place)

Reply:
I had issues with getOrElse never returning the "or else" I think, which is why I went for checking if the return is empty 

Also I wanted to put the defaults in the mache extension, but I ran into the issue where I couldn't get access to the project directory for the default values
I've now realised that I could use string replacement and placeholders, like the decompiler args, but I've got to look how it's done already

Reply:
Ive re-written argument handling to be more sane IMO